src/modules/jalousie/geometry/DreamGeometry.h

```diff
@@ -1,54 +1,222 @@
+// Copyright (C) 2020 European Spallation Source, see LICENSE file
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Encoder/decoder from PixelId to the physics EndCap parameters for
+///        the DREAM detector.
+///
+/// \brief PixelIdFromEndCapParams() encodes the PixelId from EndCap parameters
+///        The input is checked for validity and true is returned on success.
+///
+/// \brief EndCapParamsFromPixelId() decodes the EndCap parameters from the
+///        PixelId. The input is checked for validity and true is returned on
+///        success.
+///
+//===----------------------------------------------------------------------===//
+
 #include <stdint.h>
+#include <common/Assert.h>
+
+namespace DreamGeometry {
+
+/* clang-format off
+
+DREAM logical pixel layout
+--------------------------
+
+The logical pixel layout of DREAM is parameterized by Sector columns and Strip rows.
 
-// This is the main PixelId decoder function
-struct EndCapParams EndCapParamsFromPixelId(uint32_t PixelId);
+            Sector 1  Sector 2  Sector ...
```

Comment:
I would much rather have this in the ICD document - and not in ASCII ;-)

Reply:
Right. There will be an ICD. I feel that not having any explanation in the source is asking for maintenence hell.